plugins/formal-verification/formver.viper/src/org/jetbrains/kotlin/formver/viper/errors/VerificationError.kt

```diff
@@ -21,42 +19,22 @@ class VerificationError private constructor(
     }
 
     /**
-     * The [offendingNode] represents the node in Viper's AST where the error occurred.
+     * The [locationNode] represents the node in Viper's AST where the error occurred.
      * This is useful with error reporting, when we want to fetch missing information from the error,
      * and we need to reconstruct the original Kotlin's code context.
      */
-    val offendingNode: Node
-        get() = Node(result.offendingNode())
+    val locationNode: AstWrapper.Node
+        get() = AstWrapper.Node(result.offendingNode())
 
     /**
-     * The [reasonOffendingNode] represents the reason error node in Viper's AST.
-     * To understand better the role of this field, let us consider a simple example:
-     * - we call a Viper's method with defined pre-conditions.
-     * - Viper throws a [VerificationError] due to a pre-condition that is not holding.
-     * - the [offendingNode] is located at the method call site, while
-     * - the [reasonOffendingNode] is located on the failing pre-condition.
+     * The [faultProposition] represents the proposition that could not be verified in the Viper's AST.
      */
-    val reasonOffendingNode: Node
-        get() = Node(result.reason().offendingNode())
+    val faultProposition: AstWrapper.Exp
```

Comment:
nit: "fault" is a noun, so this name is hard to understand.  "unverifiable" could maybe work?  Or if you want to stick with this naming, "faulty" or "atFault" would be better.